utils/misc/dbutils.go

```diff
@@ -35,6 +40,74 @@ func GetConnectionString(c *config.Config, componentName string) string {
 	)
 }
 
+func GetDatabaseConnectionPool(
+	ctx context.Context,
+	conf *config.Config,
+	stat stats.Stats,
+	componentName string,
+) (*sql.DB, error) {
+	connStr := GetConnectionString(conf, componentName)
+	db, err := sql.Open("postgres", connStr)
+	if err != nil {
+		return nil, fmt.Errorf("Error opening connection to database: %w", err)
+	}
+	if err := db.Ping(); err != nil {
+		return nil, fmt.Errorf("Error pinging database: %w", err)
+	}
+	if err := stat.RegisterCollector(
+		collectors.NewDatabaseSQLStats(
+			"jobsdb",
+			db,
+		),
+	); err != nil {
+		return nil, fmt.Errorf("Error registering database stats collector: %w", err)
+	}
+	/*
+		TODO: find out a reasonably good value for below pool configurations
+	*/
+
+	maxConnsVar := conf.GetReloadableIntVar(40, 1, "JobsDB.maxOpenConnections")
```

Comment:
I would avoid using JobsDB, as it is an overloaded term.
```suggestion
	maxConnsVar := conf.GetReloadableIntVar(40, 1, "db.pool.maxOpenConnections")
```

Or even 

```suggestion
	maxConnsVar := conf.GetReloadableIntVar(40, 1, "db.pool.maxOpenConnections", "db."+componentName+".pool.maxOpenConnections")
```